### PR TITLE
Make usability improvements to local_steely

### PR DESCRIPTION
**Pull request by CianLR**
- File is executable
 - You now have to explicitly specify --all-plugins to try and load them all
 - Warnings if plugins don't end in .py or can't be found.
 - Gud return codes

### CODE REVIEW

File: steely/local_steely.py [hunk 1]
```diff
@@ -86,16 +92,41 @@ def load_plugins(pdir, plist=None):
     If plist is None then return all plugins found.
     """
     plugins = []
+    loaded_names = set()
     for filename, path in scan_plugins_dir(pdir):
         if plist is None or filename in plist:
             plugins.append(load_plugin(filename, path))
+            loaded_names.add(filename)
+    # Check for any missed plugins
+    if plist:
```
Comment (devoxel) [inline steely/local_steely.py]: you could do:

```
if not plist:
    return plugins
# continue other stuff
```

File: steely/local_steely.py [hunk 2]
```diff
@@ -86,16 +92,41 @@ def load_plugins(pdir, plist=None):
...
+        for filename in plist:
+            if filename not in loaded_names:
+                print("Warning: Failed to find",
```
Comment (devoxel) [inline steely/local_steely.py]: format strings would be better here

File: steely/local_steely.py [hunk 3]
```diff
@@ -86,16 +92,41 @@ def load_plugins(pdir, plist=None):
...
+                      "'" + filename + "'", "in dir",
+                      "'" + pdir + "'")
     return plugins
 
+def check_args(args):
+    if not (bool(args.p) ^ args.all_plugins):
+        print("One of -p or --all-plugins must be specified")
```
Comment (devoxel) [inline steely/local_steely.py]: at least one of?

File: steely/local_steely.py [hunk 4]
```diff
@@ -86,16 +92,41 @@ def load_plugins(pdir, plist=None):
...
+        return False
+    if args.p:
```
Comment (devoxel) [inline steely/local_steely.py]: same comment as above

File: steely/local_steely.py [hunk 5]
```diff
@@ -86,16 +92,41 @@ def load_plugins(pdir, plist=None):
...
+        for plugin_file in args.p:
+            if not plugin_file.endswith('.py'):
+                print("Warning: Plugin file",
```
Comment (devoxel) [inline steely/local_steely.py]: format strings again

File: steely/local_steely.py (hunk 3 above)
Comment (iandioch) [reply steely/local_steely.py]: It doesn't make sense to have both, does it? "at least" implies that.

**Comment by iandioch** [comment]
Will this ever be merged? Stay tuned to find out.

**Comment by byxor** [comment]
Uh ohhh, looks like a pesky programmer PEP8'd the entire codebase and left you with merge conflicts. :shy: :blush:

**Comment by iandioch** [comment]
oopsie woopsie

**Pull request merged**